presto-hive/src/main/java/io/prestosql/plugin/hive/HiveWriterFactory.java

```diff
@@ -549,6 +549,37 @@ else if (insertExistingPartitionsBehavior == InsertExistingPartitionsBehavior.ER
                 hiveWriterStats);
     }
 
+    private String computeFileName(OptionalInt bucketNumber)
+    {
+        if (isCreateTransactionalTable) {
+            // Hive is very opinionated on naming pattern for "original" files of transactional tables.
+            // As currently CTAS for transactional tables in Presto creates non-transactional ("original") files.
+            // we use special file naming mode in this case.
```

Comment:
This requirement will not go away if we impl CTAS as a write to base or delta directory.
This is just the naming required for ACID tables.

Reply:
No (or not sure). ACID tables probably also have some naming requirements. But those are different than listed here I believe. Specifically wouldn't `base_*` directory be created on first CTAS for ACID table normally?